ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,564 @@
+      res = _NNS_UINT64;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
+      res = _NNS_FLOAT32;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE:
+      res = _NNS_FLOAT64;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16:
+#ifdef FLOAT16_SUPPORT
+      res = _NNS_FLOAT16;
+      break;
+#endif
+    default:
+      nns_logw ("Tensor type not supported: %d", (gint) _type);
+      return -EINVAL;
+  }
+
+  *type = res;
+  return 0;
+}
+
+/**
+ * @brief Method to get empty object.
+ */
+tensor_filter_subplugin &
+onnxruntime_subplugin::getEmptyInstance ()
+{
+  return *(new onnxruntime_subplugin ());
+}
+
+/**
+ * @brief Method to prepare/configure onnxruntime instance.
+ */
+void
+onnxruntime_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  size_t i;
+  Ort::AllocatorWithDefaultOptions allocator;
+
+  if (!empty_model) {
+    /* Already opened */
+
+    if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+      std::cerr << "Model path is not given." << std::endl;
+      throw std::invalid_argument ("Model path is not given.");
+    }
+
+    cleanup ();
+  }
+
+  assert (model_path == nullptr);
+
+  if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
+    const std::string err_msg
+        = "Given file " + (std::string) prop->model_files[0] + " is not valid";
+    std::cerr << err_msg << std::endl;
+    cleanup ();
+    throw std::invalid_argument (err_msg);
+  }
+
+  model_path = g_strdup (prop->model_files[0]);
+
+  assert (model_path != nullptr);
+
+  if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
+    const std::string err_msg
+        = "Given file " + (std::string) prop->model_files[0] + " is not valid";
+    std::cerr << err_msg << std::endl;
+    cleanup ();
+    throw std::invalid_argument (err_msg);
+  }
+
+  /** Read a model */
+  env = Ort::Env (ORT_LOGGING_LEVEL_WARNING, "nnstreamer_onnxruntime");
+  session = Ort::Session (env, model_path, sessionOptions);
+
+  input_num_tensors = session.GetInputCount ();
+  if (input_num_tensors <= 0 || input_num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    cleanup ();
+    throw std::invalid_argument (
+        "The number of tensors required by the given model exceeds the nnstreamer tensor limit (" NNS_TENSOR_SIZE_LIMIT_STR
+        " by default).");
+  }
+  inputInfo.num_tensors = input_num_tensors;
+
+  output_num_tensors = session.GetOutputCount ();
+  if (input_num_tensors <= 0 || input_num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    cleanup ();
+    throw std::invalid_argument (
+        "The number of tensors required by the given model exceeds the nnstreamer tensor limit (" NNS_TENSOR_SIZE_LIMIT_STR
+        " by default).");
+  }
+  outputInfo.num_tensors = output_num_tensors;
+
+  // initialize input info
+  input_shapes.resize (inputInfo.num_tensors);
+  input_types.resize (inputInfo.num_tensors);
+
+  for (i = 0; i < inputInfo.num_tensors; i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < inputInfo.num_tensors; i++) {
```

the variable ```i``` is used for loops.

Reply:
Thanks. updated it.